presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlClient.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.mysql;
+
+import com.esri.core.geometry.Point;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.esri.core.geometry.ogc.OGCPoint;
+import com.facebook.presto.spi.PrestoException;
+import io.airlift.slice.Slice;
+import org.testng.annotations.Test;
+
+import java.nio.ByteBuffer;
+
+import static com.facebook.presto.geospatial.GeoFunctions.stGeomFromBinary;
+import static com.facebook.presto.plugin.mysql.MySqlClient.getAsText;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+@Test
+public class TestMySqlClient
+{
+    @Test
+    public void testValidGeometryReadMapping()
+    {
+        OGCGeometry geometry = new OGCPoint(new Point(1.0, 2.0), null);
+        ByteBuffer buffer = geometry.asBinary();
+        byte[] wkb = new byte[buffer.remaining()];
+        buffer.get(wkb);
+
+        Slice value = getAsText(stGeomFromBinary(wrappedBuffer(wkb)));
```

Comment:
You don't need to do the extra steps of creating the `wkb` byte array. Just do `Slices.wrappedBuffer(buffer)`